controllers/service_controller.go

```diff
@@ -1,109 +0,0 @@
-*/
-
-package controllers
-
-import (
-	"context"
-	"fmt"
-
-	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/client-go/tools/record"
-	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-
-	"github.com/aws/aws-application-networking-k8s/pkg/aws"
-	"github.com/aws/aws-application-networking-k8s/pkg/deploy"
-	"github.com/aws/aws-application-networking-k8s/pkg/gateway"
-	"github.com/aws/aws-application-networking-k8s/pkg/k8s"
-	lattice_runtime "github.com/aws/aws-application-networking-k8s/pkg/runtime"
-	"github.com/aws/aws-application-networking-k8s/pkg/utils/gwlog"
-)
-
-const (
-	serviceFinalizer = "service.ki8s.aws/resources"
-)
-
-type serviceReconciler struct {
-	log              gwlog.Logger
-	client           client.Client
-	scheme           *runtime.Scheme
-	finalizerManager k8s.FinalizerManager
-	eventRecorder    record.EventRecorder
-	targetsBuilder   gateway.LatticeTargetsBuilder
-	stackDeployer    deploy.StackDeployer
-	stackMashaller   deploy.StackMarshaller
-}
-
-func RegisterServiceController(
-	log gwlog.Logger,
-	cloud aws.Cloud,
-	finalizerManager k8s.FinalizerManager,
-	mgr ctrl.Manager,
-) error {
-	client := mgr.GetClient()
-	scheme := mgr.GetScheme()
-	evtRec := mgr.GetEventRecorderFor("service")
-	sr := &serviceReconciler{
-		log:              log,
-		client:           client,
-		scheme:           scheme,
-		finalizerManager: finalizerManager,
-		eventRecorder:    evtRec,
-	}
-	err := ctrl.NewControllerManagedBy(mgr).
-		For(&corev1.Service{}).
-		Complete(sr)
-	return err
-}
-
-//+kubebuilder:rbac:groups=core,resources=services,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups=core,resources=services/status,verbs=get;update;patch
-//+kubebuilder:rbac:groups=core,resources=services/finalizers,verbs=update
-//+kubebuilder:rbac:groups=core,resources=endpoints,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups=core,resources=endpoints/status,verbs=get;update;patch
-//+kubebuilder:rbac:groups=core,resources=configmaps, verbs=create;delete;patch;update;get;list;watch
-
-func (r *serviceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	r.log.Infow("reconcile", "name", req.Name)
-	recErr := r.reconcile(ctx, req)
-	if recErr != nil {
-		r.log.Infow("reconcile error", "name", req.Name, "message", recErr.Error())
-	}
-	return lattice_runtime.HandleReconcileError(recErr)
-}
-
-func (r *serviceReconciler) reconcile(ctx context.Context, req ctrl.Request) error {
-	// today, target registration is handled through the route controller
-	// if that proves not to be fast enough, we can look to do quicker
-	// target registration with the service controller
-
-	svc := &corev1.Service{}
-	if err := r.client.Get(ctx, req.NamespacedName, svc); err != nil {
-		return client.IgnoreNotFound(err)
-	}
-	if !svc.DeletionTimestamp.IsZero() {
-		r.finalizerManager.RemoveFinalizers(ctx, svc, serviceFinalizer)
```

Comment:
I think we still need this line, otherwise users cannot delete the service at all.

Reply:
Should we do a breaking change to get rid of some unnecessary code before GA...

Reply:
This finalizer has been added to all services in a cluster that have controller installed. We should almost forever try to delete this. Removing the reconciler does not even solve the problem of controller uninstallation and only making it worse.

Reply:
Ok, should we keep to just remove `"service.ki8s.aws/resources"` logic? Or remove all Finalizers, remove all finalizers could potentially have unexpected side effect?

Reply:
we should not touch other finalizers besides `service.ki8s.aws/resources`

Reply:
That being said, if we are worried about such performance a more effective move will be removing pod reconciler instead.